### ISSUE TITLE
Double dependency reporting for HTTP Requests

### ISSUE THREAD

**Issue by annanhan** [Microsoft/ApplicationInsights-node.js#349]
It looks like the remote dependencies are getting double logged for HTTP requests. I have two apps that are communicating with each other, so I can see the request getting logged once on the receiving end.

![doubledependency](https://user-images.githubusercontent.com/10620173/32618430-b6b1305a-c545-11e7-92bb-6a33e1aaf5a1.png)
It also seems like the C# library could be double logging dependencies as well.

Could this be related to: https://github.com/Microsoft/ApplicationInsights-node.js/issues/323?

Here is how I have AppInsights configured:
```
var appInsights = require('applicationinsights');
var winston = require('winston');
var expressWinston = require('express-winston');
require('le_node');

appInsights.setup(config.appInsightsKey)
      .setAutoCollectConsole(false)
      .start();
  //register common properties to global client (used for auto-collection)
  appInsights.defaultClient.commonProperties = log.commonAppInsightsProperties;
  appInsights.defaultClient.addTelemetryProcessor(telemetryProcessor);

var log = new winston.Logger({
  transports:[
    new winston.transports.Console({
      handleExceptions: true,
      prettyPrint: true,
      level: config.logLevel.console
    }),
    new winston.transports.Logentries({
      name: 'logentriesLogger',
      token: config.logentriesKey,
      handleExceptions: true,
      minLevel: config.logLevel.logentries,
      withStack: true
    }),
    new winston.transports.CustomLogger({
      level: config.logLevel.appInsights,
      handleExceptions: true
    })
  ],
  exitOnError: false
});
```

**Comment by OsvaldoRosado** [comment]
This could be related to #323. How are you making your http requests? With the built in `http` library or something else?

**Comment by annanhan** [comment]
With [request](https://www.npmjs.com/package/request).

**Comment by OsvaldoRosado** [comment]
In this case, this is definitely one of the problems described in #323, that also found the same to be happening with `request`. My current guess is that we're double collecting because of how request works internally.

Thanks for reporting this! Until this is fixed, you may be able to workaround with a telemetry processor that keeps track of what dependency calls it has seen in the current correlation context, or disable automatic dependency collection and do manual `trackDependency` calls

**Comment by OsvaldoRosado** [comment]
The double reporting is due to 300-level redirects. The request module follows these, and subsequently makes a second request. In 1.0.0 status codes were not reported correctly to showcase this difference, and that was fixed in the newly-released SDK v1.0.1.

If you don't want this redirect telemetry sent, you can now use a telemetry processor to filter out dependency telemetry for status codes indicating redirect.

Please feel free to re-open if you're experiencing something different than this in 1.0.1!
